test/integration/integration.test.ts

```diff
@@ -0,0 +1,535 @@
+        url: receiverWalletAddressUrl
+      })
+      senderWalletAddress = await c9.opClient.walletAddress.get({
+        url: senderWalletAddressUrl
+      })
+
+      expect(receiverWalletAddress.id).toBe(
+        receiverWalletAddressUrl.replace('http', 'https')
+      )
+      expect(senderWalletAddress.id).toBe(
+        senderWalletAddressUrl.replace('http', 'https')
+      )
+    })
+
+    test('Can Get Non-Existing Wallet Address', async (): Promise<void> => {
+      const notFoundWalletAddress =
+        'https://host.docker.internal:4000/accounts/asmith'
+
+      const handleWebhookEventSpy = jest.spyOn(
+        hlb.integrationServer.webhookEventHandler,
+        'handleWebhookEvent'
+      )
+
+      // Poll in case the webhook response to create wallet address is slow,
+      // but initial request may very well resolve immediately.
+      const walletAddress = await poll(
+        async () =>
+          c9.opClient.walletAddress.get({
+            url: notFoundWalletAddress
+          }),
+        (responseData) => responseData.id === notFoundWalletAddress,
+        5,
+        0.5
+      )
+
+      assert(walletAddress)
+      expect(walletAddress.id).toBe(notFoundWalletAddress)
+      expect(handleWebhookEventSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          type: WebhookEventType.WalletAddressNotFound,
+          data: expect.objectContaining({
+            walletAddressUrl: notFoundWalletAddress
+          })
+        })
+      )
+    })
+
+    test('Grant Request Incoming Payment', async (): Promise<void> => {
+      const grant = await c9.opClient.grant.request(
+        {
+          url: receiverWalletAddress.authServer
+        },
+        {
+          access_token: {
+            access: [
+              {
+                type: 'incoming-payment',
+                actions: ['create', 'read', 'list', 'complete']
+              }
+            ]
+          }
+        }
+      )
+
+      assert(!isPendingGrant(grant))
+      accessToken = grant.access_token.value
+    })
+
+    test('Create Incoming Payment', async (): Promise<void> => {
+      const now = new Date()
+      const tomorrow = new Date(now)
+      tomorrow.setDate(now.getDate() + 1)
+
+      const handleWebhookEventSpy = jest.spyOn(
+        hlb.integrationServer.webhookEventHandler,
+        'handleWebhookEvent'
+      )
+
+      incomingPayment = await c9.opClient.incomingPayment.create(
+        {
+          url: receiverWalletAddress.resourceServer,
+          accessToken
+        },
+        {
+          walletAddress: receiverWalletAddressUrl.replace('http', 'https'),
+          incomingAmount: {
+            value: '100',
+            assetCode: receiverWalletAddress.assetCode,
+            assetScale: receiverWalletAddress.assetScale
+          },
+          metadata: { description: 'Free Money!' },
+          expiresAt: tomorrow.toISOString()
+        }
+      )
+
+      // Delay gives time for webhook to be received
```

Comment:
could we also poll here instead?

Reply:
Makes sense, we could even decrease the polling frequency (although its very much not a big deal)